rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -49,28 +71,113 @@
     private FlowNodeUtil() {
     }
 
-    public static long getNodeExecDuration(FlowNode node) {
-        List<FlowNode> childNodes = getChildNodes(node);
-        if (!childNodes.isEmpty()) {
-            long startTime = TimingAction.getStartTime(node);
-            long endTime = TimingAction.getStartTime(childNodes.get(childNodes.size() - 1));
+    public abstract static class CacheExtensionPoint implements ExtensionPoint {
+        public abstract Cache<String,List<FlowNode>> getExecutionCache();
+        public abstract Cache<String, RunExt> getRunCache();
+        public abstract Cache<FlowNode,String> getExecNodeNameCache();
+    }
 
-            if (startTime == 0) {
-                // The node is running and the time has not been marked on it yet.  Return 0 as the duration for now.
-                return 0;
+    // Used in testing where Jenkins is not running yet
+    private static final List<CacheExtension> FALLBACK_CACHES = Arrays.asList(new CacheExtension());
```

Comment:
The list is not really a constant, since `Arrays.asList` returns an `ArrayList` that can be added and removed from, but since it's `private` I see no real harm with it, otherwise it would need to be wrapped with `Collections.unmodifiableList`